### PR TITLE
bullet trimesh caching

### PR DESCRIPTION
**Pull request by lealzhan**
Re: #https://github.com/cocos/3d-tasks/issues/12949

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/physics/bullet/bullet-triangle-mesh.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+ */
+
+import { bt, EBulletType } from './instantiated';
+import { Mesh } from '../../3d/assets';
+import { cocos2BulletTriMesh } from './bullet-utils';
+
+export class BulletTriangleMesh {
+    private static readonly BulletTriangleMeshMap = new Map<number, BulletTriangleMesh>();
```
Comment (yiwenxue) [inline cocos/physics/bullet/bullet-triangle-mesh.ts:31]: BulletTriangleMeshMap is a private member, use _BulletTriangleMeshMap instead

Comment (lealzhan) [reply cocos/physics/bullet/bullet-triangle-mesh.ts:31]: ok

File: cocos/physics/bullet/shapes/bullet-trimesh-shape.ts [hunk 2]
```diff
@@ -49,13 +52,20 @@ export class BulletTrimeshShape extends BulletShape implements ITrimeshShape {
                 const btTriangleMesh = this._getBtTriangleMesh(mesh);
                 if (this.collider.convex) {
                     this._impl = bt.ConvexTriangleMeshShape_new(btTriangleMesh);
-                } else {
-                    this._impl = bt.BvhTriangleMeshShape_new(btTriangleMesh, true, true);
+                } else if (BulletTrimeshShape.mapTrimesh2btBVHMeshShape[mesh.hash] == null) { // triangle mesh and bvh is not built
+                    const bvhTrimesh = bt.BvhTriangleMeshShape_new(btTriangleMesh, true, true);
+                    BulletTrimeshShape.mapTrimesh2btBVHMeshShape[mesh.hash] = bvhTrimesh;
+                    const scale = new Vec3(1);
```
Comment (yiwenxue) [inline cocos/physics/bullet/shapes/bullet-trimesh-shape.ts]: > const scale = new Vec3(1);

redundant code ?

Comment (lealzhan) [reply cocos/physics/bullet/shapes/bullet-trimesh-shape.ts]: yes

**Review by yiwenxue — APPROVED** [review]

**Review by SantyWang — APPROVED** [review]

**Pull request merged**